pkg/yurthub/kubernetes/rest/config_test.go

```diff
@@ -0,0 +1,186 @@
+clusters:
+- cluster:
+    certificate-authority-data: temp
+    server: https://10.10.10.113:6443
+  name: default-cluster
+contexts:
+- context:
+    cluster: default-cluster
+    namespace: default
+    user: default-auth
+  name: default-context
+current-context: default-context
+kind: Config
+preferences: {}
+users:
+- name: default-auth
+  user:
+    client-certificate: /tmp/pki/yurthub-current.pem
+    client-key: /tmp/pki/yurthub-current.pem
+`)
+	testDir = "/tmp/pki/"
+)
+
+func TestGetRestConfig(t *testing.T) {
+	remoteServers := map[string]int{"https://10.10.10.113:6443": 2}
+	u, _ := url.Parse("https://10.10.10.113:6443")
+	fakeHealthchecker := healthchecker.NewFakeChecker(true, remoteServers)
+	dStorage, err := disk.NewDiskStorage(testDir)
+	defer func() {
+		if err := os.RemoveAll(testDir); err != nil {
+			t.Errorf("Unable to clean up test directory %q: %v", testDir, err)
+		}
+	}()
+
+	// store the kubelet ca file
+	caFile := filepath.Join(testDir, "ca.crt")
+	if err := dStorage.Create("ca.crt", certificatePEM); err != nil {
+		t.Fatalf("Unable to create the file %q: %v", caFile, err)
+	}
+
+	// store the kubelet-pair.pem file
+	pairFile := filepath.Join(testDir, "kubelet-pair.pem")
+	pd := bytes.Join([][]byte{certificatePEM, keyPEM}, []byte("\n"))
+	if err := dStorage.Create("kubelet-pair.pem", pd); err != nil {
+		t.Fatalf("Unable to create the file %q: %v", pairFile, err)
+	}
+
+	// store the yurthub-current.pem
+	yurthubCurrent := filepath.Join(testDir, "yurthub-current.pem")
+	if err := dStorage.Create("yurthub-current.pem", pd); err != nil {
+		t.Fatalf("Unable to create the file %q: %v", yurthubCurrent, err)
+	}
+
+	// store the yurthub.conf
+	yurthubConf := filepath.Join(testDir, "yurthub.conf")
+	if err := dStorage.Create("yurthub.conf", yurthubCon); err != nil {
+		t.Fatalf("Unable to create the file %q: %v", yurthubConf, err)
+	}
+
+	// set the YurtHubConfiguration
+	cfg := &config.YurtHubConfiguration{
+		RootDir:               testDir,
+		RemoteServers:         []*url.URL{u},
+		KubeletRootCAFilePath: caFile,
+		KubeletPairFilePath:   pairFile,
+	}
+
+	tests := []struct {
+		desc string
+		mode string
+	}{
+		{desc: "kubelet mode", mode: "kubelet"},
+		{desc: "hubself mode", mode: "hubself"},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			cfg.CertMgrMode = tt.mode
+			var certMgr interfaces.YurtCertificateManager
+			if tt.mode == "kubelet" {
+				certMgr, err = kubelet.NewKubeletCertManager(cfg, 10*time.Second)
+			} else if tt.mode == "hubself" {
+				cfg.NodeName = "node"
+				certMgr, err = hubself.NewYurtHubCertManager(cfg)
+			}
+
+			if err != nil {
+				t.Errorf("failed to create %s certManager: %v", tt.mode, err)
+			}
+
+			rcm, err := NewRestConfigManager(cfg, certMgr, fakeHealthchecker)
+			if err != nil {
+				t.Errorf("failed to create RestConfigManager: %v", err)
+			}
+
+			var rc *rest.Config
+			if tt.mode == "hubself" {
+				rc = rcm.getHubselfRestConfigSkipCurrent()
```

Comment:
I think we need invoke `GetRestConfig()` here and then verify the return value.

Reply:
The unit test has been corrected, and the `GetRestConfig()` function will be called here.